### PR TITLE
Orca: add validation in NCF examples

### PR DESCRIPTION
**Pull request by hamham223**
### 1. Why the change?

The current examples do not involve validation in `fit()`. Use test data as validation data.

### 2. How to test?
- [ ] N/A
- [x] Unit test
- [ ] Application test
- [ ] Document test
- [ ] ...

**The PR should not be merged only after https://github.com/intel-analytics/BigDL/pull/7182 is finished**

### CODE REVIEW

**Comment by hamham223** [comment]
Here is some preview:

+ pytorch
outcome
+ tensorflow
outcome_tf

note, `tf_runner.py` has some bugs so only the last epoch is returned

File: python/orca/tutorial/NCF/pytorch_train_dataloader.py [hunk 1]
```diff
@@ -91,7 +91,16 @@ def optimizer_creator(model, config):
                                    "lr": 0.01,
                                    "model": "NeuMF-end",
                                    "sparse_feats_embed_dims": 8})
-est.fit(data=train_loader_func, epochs=2, batch_size=10240, callbacks=callbacks)
+history = est.fit(data=train_loader_func,
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_train_dataloader.py:94]: do not call it history, call it train_stats instead?

File: python/orca/tutorial/NCF/pytorch_train_dataloader.py [hunk 2]
```diff
@@ -91,7 +91,16 @@ def optimizer_creator(model, config):
...
+                  batch_size=10240,
+                  callbacks=callbacks)
+print("Train results:")
+for epoch in history:
+    for e in epoch:
+        print("{}: {}".format(e, epoch[e]))
+    print()
 
 
 # Step 5: Distributed evaluation of the trained model
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_train_dataloader.py:106]: change result to eval_stats?

Comment (hamham223) [reply python/orca/tutorial/NCF/pytorch_train_dataloader.py:106]: updated

File: python/orca/tutorial/NCF/pytorch_train_dataloader.py [hunk 3]
```diff
@@ -91,14 +91,23 @@ def optimizer_creator(model, config):
...
                                    "model": "NeuMF-end",
                                    "sparse_feats_embed_dims": 8})
-est.fit(data=train_loader_func, epochs=2, batch_size=10240, callbacks=callbacks)
+train_stats = est.fit(data=train_loader_func,
+                      validation_data=test_loader_func,
+                      epochs=2,
+                      batch_size=10240,
+                      callbacks=callbacks)
+print("Train results:")
+for epoch in train_stats:
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_train_dataloader.py:100]: epoch -> epoch_stats

File: python/orca/tutorial/NCF/pytorch_train_dataloader.py [hunk 4]
```diff
@@ -91,14 +91,23 @@ def optimizer_creator(model, config):
...
+    for e in epoch:
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_train_dataloader.py:101]: maybe change to `for k, v in epoch_stats.items()`?

File: python/orca/tutorial/NCF/pytorch_train_dataloader.py [hunk 5]
```diff
@@ -91,14 +91,23 @@ def optimizer_creator(model, config):
...
 
 
 # Step 5: Distributed evaluation of the trained model
-result = est.evaluate(data=test_loader_func, batch_size=10240)
+eval_stats = est.evaluate(data=test_loader_func, batch_size=10240)
 print("Evaluation results:")
-for r in result:
-    print("{}: {}".format(r, result[r]))
+for e in eval_stats:
+    print("{}: {}".format(e, eval_stats[e]))
```
Comment (hkvision) [inline python/orca/tutorial/NCF/pytorch_train_dataloader.py:110]: also change to .items for eval...

File: python/orca/tutorial/NCF/tf_train_spark_dataframe.py [hunk 6]
```diff
@@ -74,24 +74,29 @@ def model_creator(config):
...
-        steps_per_epoch=train_steps,
-        callbacks=callbacks)
+train_stats = est.fit(train_df,
+                      epochs=2,
+                      batch_size=batch_size,
+                      feature_cols=feature_cols,
+                      label_cols=label_cols,
+                      steps_per_epoch=train_steps,
+                      validation_data=test_df,
+                      validation_steps=2,
```
Comment (hkvision) [inline python/orca/tutorial/NCF/tf_train_spark_dataframe.py:84]: 2 is wrong, you need to calculate validation steps

File: python/orca/tutorial/NCF/tf_train_spark_dataframe.py [hunk 7]
```diff
@@ -74,24 +74,29 @@ def model_creator(config):
...
-                      num_steps=val_steps)
+eval_stats = est.evaluate(test_df,
+                          feature_cols=feature_cols,
+                          label_cols=label_cols,
+                          batch_size=batch_size,
+                          num_steps=val_steps)
 print("Evaluation results:")
-for r in result:
-    print("{}: {}".format(r, result[r]))
+for e in eval_stats:
```
Comment (hkvision) [inline python/orca/tutorial/NCF/tf_train_spark_dataframe.py:98]: same as above

**Pull request merged**